scr/main/java/bittercode/model/Book.java

```diff
@@ -0,0 +1,65 @@
+package com.bittercode.model;
+
+import java.io.Serializable;
+
+public class Book implements Serializable {
+
+    private String barcode;
+    private String name;
+    private String author;
+    private double price;
+    private int quantity;
+
+    public Book(String barcode, String name, String author, double price, int quantity) {
+        this.barcode = barcode;
+        this.name = name;
+        this.author = author;
+        this.setPrice(price);
+        this.quantity = quantity;
+    }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider input validation in constructor</b></div>
<div id="fix">

Consider validating input parameters in the constructor to ensure data integrity. For example, `price` and `quantity` should not be negative.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -13,7 +13,11 @@
     public Book(String barcode, String name, String author, double price, int quantity) {
 +        if (price < 0 || quantity < 0) {
 +            throw new IllegalArgumentException("Price and quantity must be non-negative");
 +        }
         this.barcode = barcode;
         this.name = name;
         this.author = author;
         this.setPrice(price);
         this.quantity = quantity;
     }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2514333017>#855223</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged